custom_components/mqtt_vacuum_camera/camera.py

```
@@ -236,6 +241,134 @@
+            str: The full path to the saved image or None if the download fails.
+        """
+        # Ensure the storage path exists
+        os.makedirs(storage_path, exist_ok=True)
+
+        obstacle_file = os.path.join(storage_path, filename)
+
+        async def blocking_download():
+            """Run the blocking download in a separate thread."""
+            try:
+                async with aiohttp.ClientSession() as session:
+                    async with session.get(url) as response:
```

Ruff (0.8.0)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)